app/models/team.rb

```diff
@@ -46,16 +46,14 @@ class Team < ActiveRecord::Base
 
   scope :by_season, ->(year_or_season) do
     case year_or_season
-      when Integer
-        joins(:season).where('seasons.name': year_or_season)
-      when String
-        joins(:season).where('seasons.name': year_or_season)
-      when Season
-        where(season: year_or_season)
-      else
-        raise
-      end
+    when Integer, String
+      joins(:season).where('seasons.name': year_or_season)
+    when Season
+      where(season: year_or_season)
+    else
+      none
```

Comment:
🎉  haven't seen this before, but it makes HUGE sense here 👍  (`[]` would not allow to chain scopes...)